src/filters/path.rs

```diff
@@ -208,14 +218,17 @@ where
     fn filter(&self, _: Internal) -> Self::Future {
         route::with(|route| {
             let p = self.0.as_ref();
-            future::ready(with_segment(route, |seg| {
+            future::ready(with_segments(route, |seg| {
                 tracing::trace!("{:?}?: {:?}", p, seg);
 
-                if seg == p {
-                    Ok(())
-                } else {
-                    Err(reject::not_found())
-                }
+                (
+                    p.len(),
+                    if seg.starts_with(p) {
```

Comment:
I think this will could allow false matches. For example:

```rust
let req = warp::test::request().path("/foobar");
let p = warp::path!("foo" / ..);
assert!(!req.matches(&p).await);
```

This panics because it advances the internal segment index to one that isn't a `/`.